### PR TITLE
Topic4

### PR DESCRIPTION
**Pull request by Marcosgreppi**

### CODE REVIEW

File: 04-HTML5-APIs/4-1.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  writeToScreen("Error");
+}
+
+function onMessage(evt) {
+  writeToScreen("<span style="color: blue;">RESPONSE: " + evt.data+"</span>");
+  websocket.close();
+}
+
+function onError(evt) {
+  writeToScreen("<span style="color: red;">ERROR:</span> " + evt.data);
```
Comment (afiorenza) [inline 04-HTML5-APIs/4-1.js]: The style should go on a css file.

File: 04-HTML5-APIs/4-1.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+let output;
+
+function init() {
+  output = document.getElementById("output");
+  testWebSocket();
+}
+
+function testWebSocket() {
+  websocket = new WebSocket(wsUri);
+  websocket.onopen = function(evt) { onOpen(evt) };
```
Comment (afiorenza) [inline 04-HTML5-APIs/4-1.js]: it is not necessary to declare a function to just call a function. You could directly call onOpen function. The same with following lines.

File: 04-HTML5-APIs/4-1.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+  websocket.onclose = function(evt) { onClose(evt) };
+  websocket.onmessage = function(evt) { onMessage(evt) };
+  websocket.onerror = function(evt) { onError(evt) };
+}
+
+function onOpen(evt) {
```
Comment (afiorenza) [inline 04-HTML5-APIs/4-1.js]: `evt` is not a descriptive name. Please refactor this.

File: 04-HTML5-APIs/4-1.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  writeToScreen("Works");
+  doSend("WebSocket works");
+}
+
+function onClose(evt) {
+  writeToScreen("Error");
+}
+
+function onMessage(evt) {
+  writeToScreen("<span style="color: blue;">RESPONSE: " + evt.data+"</span>");
```
Comment (afiorenza) [inline 04-HTML5-APIs/4-1.js]: The style should go on a css file. Also you could use template literals.

File: 04-HTML5-APIs/4-1.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  writeToScreen("<span style="color: red;">ERROR:</span> " + evt.data);
+}
+
+function doSend(message) {
+  writeToScreen("SENT: " + message);
+  websocket.send(message);
+}
+
+function writeToScreen(message) {
+  let pre = document.createElement("p");
```
Comment (afiorenza) [inline 04-HTML5-APIs/4-1.js]: What is pre? Refactor this with a more descriptive name.

File: 04-HTML5-APIs/4-1.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  pre.style.wordWrap = "break-word";
```
Comment (afiorenza) [inline 04-HTML5-APIs/4-1.js]: Style should go on css file.

File: 04-HTML5-APIs/4-2.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
+window.onload = function(){
+  let canvas = document.getElementById('canvas');
+  console.log(document.getElementById('canvas'))
```
Comment (afiorenza) [inline 04-HTML5-APIs/4-2.js]: Remember to remove console.logs. Are just for debugging.

**Review by afiorenza — CHANGES_REQUESTED** [review]
Good job! just a few changes

File: 04-HTML5-APIs/4-1.js (hunk 3 above)
Comment (Marcosgreppi) [reply 04-HTML5-APIs/4-1.js]: fixed

File: 04-HTML5-APIs/4-2.js (hunk 7 above)
Comment (Marcosgreppi) [reply 04-HTML5-APIs/4-2.js]: fixed

File: 04-HTML5-APIs/4-1.js (hunk 2 above)
Comment (Marcosgreppi) [reply 04-HTML5-APIs/4-1.js]: fixed

File: 04-HTML5-APIs/4-1.js (hunk 5 above)
Comment (Marcosgreppi) [reply 04-HTML5-APIs/4-1.js]: fixed

**Review by afiorenza — APPROVED** [review]

**Pull request merged**